async_simple/util/Queue.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022, Alibaba Group Holding Limited;
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* The file implements a simple thread pool. The scheduling strategy is a simple
+ * random strategy. Simply, ThreadPool would create n threads. And for a task
+ * which is waiting to be scheduled. The ThreadPool would choose a thread for
+ * this task randomly.
+ *
+ * The purpose of ThreadPool is for testing. People who want to use async_simple
+ * in actual development should implement/use more complex executor.
+ */
+#ifndef ASYNC_SIMPLE_QUEUE_H
+#define ASYNC_SIMPLE_QUEUE_H
+
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <utility>
+
+namespace async_simple::util {
+
+template <typename T>
+class Queue {
+public:
+    Queue() = default;
+
```

Comment:
From the implementation, I think we should add a static_assert: static_assert(std:: is_move_assignable_v<T>, xxxx);

Reply:
I think the ```Queue() = default;``` can be removed, because of it has mutex and it is non-copyable.